students/rhamersky/Lesson_3/mailroom_part1.py

```diff
@@ -0,0 +1,269 @@
+        lst_table.sort(reverse=True)
+
+        index = 0
+        # Formatting the money elements and changing all int() & float() data to str() data
+        # to use for creating the chart.
+        for row in range(len(lst_table)):
+            lst_table[row][index] = "${:,.2f}".format(lst_table[row][index])  # --> Total amount for gifts
+            lst_table[row][index + 1] = "${:,.2f}".format(lst_table[row][index + 1])  # --> Avg. gift amount
+            lst_table[row][index + 2] = str(lst_table[row][index + 2])  # --> Number of gifts
+        # Reverses list
+        for row in lst_table:
+            row.reverse()
+
+        lst_headers = ["Donor_Name", "Num_Gifts", "Average_Gift", "Total_Given"]  # --> List of headers for the chart.
+        # Calculating the max length of all the elements.
+        length1 = max(len(word) for word in lst_headers) + 10  # --> Max length for the header list
+        length2 = max(len(word) for row in lst_table for word in row)  # --> Max length for the donor table
+        # Calculates the max between the header list and the donor list
+        max_length = max(length1, length2)
+        # Creates the header for the chart
+        for heading in lst_headers:
+            print(("{0:{padding}}".format(heading, padding=max_length)), end=" ")
+        print()
+        # Creates the donor rows for the chart.
+        for row in lst_table:
+            print(("{0:{padding}}".format(row[0], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[1], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[2], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[3], padding=max_length)), end=" ")
+            print()
+        return
+
+
+def donor_history(donor_name, donation_amount):
+    '''
+    Updates donor's donation history or creates a new donor.
+    :param donor_name: The name of the donor.
+    :param donation_amount: The amount of money the donor donated.
+    :return: This method does not return anything. Return is just used to clearly show where this method ends.
+    '''
+    lst_donor_name = []
+    global tup_donor_names
+    # Creates a donor name list
+    for row in range(len(tup_donor_names)):
+        lst_donor_name.append(tup_donor_names[row][0])
+    # Checks to see if donor has given before and updates history
+    if donor_name in lst_donor_name:
+        index = lst_donor_name.index(donor_name)  # --> Finds index of donor in  list
+        tup_donor_names[index][1] = tup_donor_names[index][
+                                          1] + 1  # --> Increase amount of times donor donated
+        tup_donor_names[index].append(donation_amount)  # --> Adds donation to history
+    # Adds new donor to the donor list
+    else:
+        tup_new_donor = ([donor_name, 1, donation_amount],)
+        tup_donor_names += tup_new_donor
+    return
```

Comment:
don't need the returns in void functions unless you're exiting early (it is not pythonic)